homeassistant/components/wsdot/config_flow.py

```diff
@@ -0,0 +1,166 @@
+            return self.async_abort(reason="cannot_connect")
+
+        subentries = []
+        for route in import_info[CONF_TRAVEL_TIMES]:
+            maybe_travel_time = [
+                tt
+                for tt in travel_time_routes
+                # old platform configs could store the id as either a str or an int
+                if str(tt.TravelTimeID) == str(route[CONF_ID])
+            ]
+            if not maybe_travel_time:
+                _LOGGER.error(
+                    "Found legacy WSDOT travel_time that does not describe a valid travel_time route (%s)",
+                    route,
+                )
+                continue
+            travel_time = maybe_travel_time[0]
+            route_name = travel_time.Name
+            unique_id = "_".join(travel_time.Name.split())
+            subentries.append(
+                ConfigSubentry(
+                    subentry_type=SUBENTRY_TRAVEL_TIMES,
+                    unique_id=unique_id,
+                    title=route_name,
+                    data=MappingProxyType(
+                        {CONF_NAME: travel_time.Name, CONF_ID: travel_time.TravelTimeID}
+                    ),
+                ).as_dict()
+            )
+
+        return self.async_create_entry(
+            title=DOMAIN,
+            data={
+                CONF_API_KEY: import_info[CONF_API_KEY],
+            },
+            subentries=subentries,
+        )
+
+    @classmethod
+    @callback
+    def async_get_supported_subentry_types(
+        cls, config_entry: ConfigEntry
+    ) -> dict[str, type[ConfigSubentryFlow]]:
+        """Return subentries supported by wsdot."""
+        return {SUBENTRY_TRAVEL_TIMES: TravelTimeSubentryFlowHandler}
+
+
+class TravelTimeSubentryFlowHandler(ConfigSubentryFlow):
+    """Handle subentry flow for adding WSDOT Travel Times."""
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> SubentryFlowResult:
+        """Add a new Travel Time subentry."""
+        runtime = self._get_entry().runtime_data
+        if runtime is None:
+            raise ConfigEntryError("WSDOT entry has no runtime_data")
+        travel_times = await runtime.wsdot_travel_times.get_all_travel_times()
+        if user_input is not None:
+            route = [
+                {CONF_NAME: tt.Name, CONF_ID: tt.TravelTimeID}
+                for tt in travel_times
+                if tt.Name == user_input[CONF_NAME]
+            ][0]
+            name = route[CONF_NAME]
+            unique_id = "_".join(name.split())
+            return self.async_create_entry(title=name, unique_id=unique_id, data=route)
+
+        names = sorted(tt.Name for tt in travel_times)
+        return self.async_show_form(
+            step_id=SOURCE_USER,
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_NAME): selector(
+                        {"select": {"options": names, "mode": "dropdown"}}
+                    ),
```

Comment:
for this you can use a SelectSelector
```py
        options = [
            SelectOptionDict(label=station.names["long"], value=code)
            for code, station in self.stations.items()
        ]
        return self.async_show_form(
            step_id="user",
            data_schema=vol.Schema(
                {
                    vol.Required(CONF_NAME): str,
                    vol.Required(CONF_FROM): SelectSelector(
                        SelectSelectorConfig(options=options, sort=True),
                    ),
```
it can sort itself :)